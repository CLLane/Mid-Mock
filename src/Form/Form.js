import React, { Component } from 'react';

export default class Form extends Component {
  constructor() {
    super();
    this.state = {
      img: 'https://s7d2.scene7.com/is/image/DesignWithinReach/PD_2514014_MAIN?$main$',
      name: '',
      description: '',
      price: ''
    }
  }

updateInputs = (e) => {
  this.setState({ [e.target.name]: e.target.value})
}

  render() {
    return (
      <form>
        <input 
        type='text'
        name='name'
        placeholder='Product Name'
        value={this.state.name}
        onChange={(e) => this.updateInputs(e)}>
        </input>
        <input
          type='text'
          name='description'
          placeholder='Product Description'
          value={this.state.description}
          onChange={(e) => this.updateInputs(e)}>
        </input>
        <input
          type='text'
          name='price'
          placeholder='Product price'
          value={this.state.price}
          onChange={(e) => this.updateInputs(e)}>
        </input>
        <button></button>
      </form>
    )
  }
}