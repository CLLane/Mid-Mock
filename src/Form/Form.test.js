import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Form from './Form.js';

describe('Form', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Form createCard={jest.fn()}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should reset the inputs to empty strings when reset input is called', () => {
    wrapper.instance().setState({
      img: 'https://s7d2.scene7.com/is/image/DesignWithinReach/PD_2514014_MAIN?$main$',
      name: 'Chris',
      description: 'Hoping this test is going to pass',
      price: 'To Damn High'
    })

    wrapper.instance().resetInputs()

    expect(wrapper.state()).toEqual({
      img: 'https://s7d2.scene7.com/is/image/DesignWithinReach/PD_2514014_MAIN?$main$',
      name: '',
      description: '',
      price: ''
    })
  })
  
})