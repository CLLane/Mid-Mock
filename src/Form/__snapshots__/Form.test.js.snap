// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Product Name"
    type="text"
    value=""
  />
  <input
    name="description"
    onChange={[Function]}
    placeholder="Product Description"
    type="text"
    value=""
  />
  <input
    name="price"
    onChange={[Function]}
    placeholder="Product price"
    type="text"
    value=""
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    Add Purchase
  </button>
</form>
`;
